### PR TITLE
fix:allow clear button in tree select widget

### LINKED ISSUE #16522: [Bug]: The "x" icon in TreeSelect widget is visible even when no options are selected
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

On enabling the `Allow Clearing Value` property we get the `x` icon in the widget which ideally should be visible only when there is any option selected in the widget. But right now we see it even when there is no option selected. Also when we set an `onOptionChange` event and click on the `x` button, the event gets triggered.

**Note:** This bug is present only in `Release` and not in `Production`, hence marking it as `Regression`.

https://www.loom.com/share/7662751f8550499cbe65b86a7fa624ca

### Steps To Reproduce

1. D&D a TreeSelect widget.
2. Enable the `Allow Clearing Value` property.
3. Now click on the `x` icon to clear the selected option.

**Expected:** The `x` icon should disappear once the selected option is cleared.

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description
Fix Allow Clear button in TreeSelect

Fixes #16522

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/widgets/SingleSelectTreeWidget/component/index.tsx [hunk 1]
```diff
@@ -304,7 +308,7 @@ function SingleSelectTreeComponent({
           treeDefaultExpandAll={expandAll}
           treeIcon
           treeNodeFilterProp="label"
-          value={filter ? "" : value} // value should empty when filter value exist otherwise dropdown flickers #12714
+          value={value}
```
Comment (keyurparalkar) [inline app/client/src/widgets/SingleSelectTreeWidget/component/index.tsx:311]: @Tooluloope can we check if the issue: https://github.com/appsmithorg/appsmith/pull/12979 occurs with this change?

Comment (Tooluloope) [reply app/client/src/widgets/SingleSelectTreeWidget/component/index.tsx:311]: Also tested in DP

**Review by keyurparalkar — DISMISSED** [review]
LGTM!

**Comment by chandannkumar** [comment]
@Tooluloope 
- Placeholder is not visible on the Tree select widget. 
https://www.loom.com/share/1ba8d0e1c341411fb45159d8c9809b77

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/Select/Single_Select_Tree_spec.js [hunk 2]
```diff
@@ -8,107 +8,129 @@ describe("Single Select Widget Functionality", function() {
...
-    cy.treeSelectDropdown("Light Blue");
-    // Check if isDirty is set to true
-    cy.get(".t--widget-textwidget").should("contain", "true");
-    // Change defaultText
-    cy.openPropertyPane("singleselecttreewidget");
-    cy.updateCodeInput(".t--property-control-defaultselectedvalue", "RED");
-    // Check if isDirty is reset to false
-    cy.get(".t--widget-textwidget").should("contain", "false");
-  });
+  // it("1. Check isDirty meta property", function() {
```
Comment (keyurparalkar) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/Select/Single_Select_Tree_spec.js:11]: @Tooluloope Why are we commenting these test cases? Can we remove the comments?

Comment (Tooluloope) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/Select/Single_Select_Tree_spec.js:11]: Forgot to revert after writing the tests, thanks for catching that

**Review by keyurparalkar — APPROVED** [review]

**Pull request merged**